src/window/macos.rs

```diff
@@ -175,3 +176,52 @@ impl MacosWindowFeature {
         }
     }
 }
+
+pub struct Menu {
+    menu_added: bool,
+}
+
+impl Default for Menu {
+    fn default() -> Self {
+        Menu { menu_added: false }
+    }
+}
+
+impl Menu {
+    pub fn ensure_menu_added(&mut self, ev: &Event<UserEvent>) {
+        if let Event::WindowEvent { event: WindowEvent::Focused(_), .. } =  ev {
+            if !self.menu_added {
+                self.add_window_menu();
+                self.menu_added = true;
+            }
+        }
+    }
+
+    #[cfg(target_os = "macos")]
+    fn add_window_menu(&self) {
+        use cocoa::appkit::{NSApplication, NSMenu, NSMenuItem};
```

Comment:
This should probably use icrate instead of cocoa, like 
* https://github.com/neovide/neovide/pull/2351